google-ads/src/main/java/com/google/ads/googleads/lib/utils/FieldMasks.java

```diff
@@ -119,6 +123,65 @@ private static void compare(
     }
   }
 
+  /**
+   * Gets the field value of a message from a field mask path.
+   *
+   * @param fieldMaskPath The field mask path.
+   * @param entity The entity to retrieve values from.
+   * @return the field Object.
+   */
+  public static Object getFieldValue(String fieldMaskPath, Message entity) {
+    List<String> fieldMaskParts = new ArrayList<>(Arrays.asList(fieldMaskPath.split("\\.")));
```

Comment:
Removing the first element of an ArrayList can get expensive because it has to shift everything back a slot in the underlying array. It'd be more efficient to use a `LinkedList` here, and declare it as a `Deque`.

Then you can just do:

```
while (!fieldMaskParts.isEmpty()) {
   String fieldMaskPart = fieldMaskParts.remove();
   ...
}
```